src/libraries/Microsoft.PowerFx.Connectors/ODataVisitor.cs

```diff
@@ -0,0 +1,237 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Specialized;
+using System.Globalization;
+using Microsoft.PowerFx.Core.IR;
+using Microsoft.PowerFx.Core.IR.Nodes;
+using Microsoft.PowerFx.Core.IR.Symbols;
+using Microsoft.PowerFx.Core.Texl.Builtins;
+using Microsoft.PowerFx.Interpreter;
+using Microsoft.PowerFx.Types;
+
+namespace Microsoft.PowerFx.Connectors
+{
+    internal class ODataVisitor : IRNodeVisitor<string, DelegationRunContext>
+    {
+        internal static readonly ODataVisitor I = new ODataVisitor();
+
+        public override string Visit(TextLiteralNode node, DelegationRunContext runContext)
+        {
+            return $"'{node.LiteralValue}'";
+        }
+
+        public override string Visit(NumberLiteralNode node, DelegationRunContext runContext)
+        {
+            return node.LiteralValue.ToString(CultureInfo.InvariantCulture);
+        }
+
+        public override string Visit(BooleanLiteralNode node, DelegationRunContext runContext)
+        {
+            return node.LiteralValue.ToString(CultureInfo.InvariantCulture);
+        }
+
+        public override string Visit(ColorLiteralNode node, DelegationRunContext runContext)
+        {
+            throw new NotDelegableException();
+        }
+
+        public override string Visit(RecordNode node, DelegationRunContext runContext)
+        {
+            throw new NotDelegableException();
+        }
+
+        public override string Visit(ErrorNode node, DelegationRunContext runContext)
+        {
+            throw new NotDelegableException();
+        }
+
+        public override string Visit(LazyEvalNode node, DelegationRunContext runContext)
+        {
+            return node.Child.Accept(this, runContext);
+        }
+
+        public override string Visit(CallNode node, DelegationRunContext runContext)
+        {
+            // Special cases for functions supported by OData
+            if (node.Function is NotFunction)
+            {
+                return $"NOT({node.Args[0].Accept(this, runContext)})";
```

Comment:
From OData Spec
not operator is lowercase: https://docs.oasis-open.org/odata/odata/v4.01/odata-v4.01-part2-url-conventions.html#sec_Not

As are the others (startswith, endswith,...)

The implementation that you're testing against may not care about case, but we should match the spec for these.

Reply:
updated as lowercase, thanks!